src/mono/mono/mini/aot-compiler.c

```diff
@@ -14355,6 +14382,50 @@ add_interp_in_wrapper_for_sig (MonoAotCompile *acfg, MonoMethodSignature *sig)
 	add_method (acfg, wrapper);
 }
 
+static gboolean
+mono_aot_add_direct_pinvoke (MonoAotCompile *acfg, char *dpi)
+{
+	char *direct_pinvoke = g_strdup (dpi);
+	if (!direct_pinvoke)
+		return TRUE;
+
+	if (direct_pinvoke[0] == '#')
+		return TRUE;
+
+	g_strstrip (direct_pinvoke);
+	if (!direct_pinvoke)
+		return TRUE;
+
+	char **direct_pinvoke_split = g_strsplit (direct_pinvoke, "!", -1);
+	char *library_name = g_strdup (direct_pinvoke_split[0]);
```

Comment:
Can fail and return NULL.

Reply:
Thanks, added a null check for both `library_name` and `entrypoint_name`